### DISCUSSION TITLE
Questions on how the path segment's `mapped_locations` feature works.

### DISCUSSION THREAD

**Discussion by osterhoutan** [JanDeDobbeleer/oh-my-posh#690] [Troubleshoot] [answered]
### Prerequisites

- [X] I have read and understand the `CONTRIBUTING` guide
- [X] I looked for duplicate issues before submitting this one

### Description

This is less of a Bug and more of a question on how things are expected to function with this particular feature (the path segment's `mapped_locations` property) as the [doc page](https://ohmyposh.dev/docs/path) on this feature is is kind of vague, and the only other thing I could find on it was this resolved issue: #187 which only seems loosely related until a community follow up ([link](https://github.com/JanDeDobbeleer/oh-my-posh/issues/187#issuecomment-738625964)) that requested that the Regex feature also be added to the file mappings of the path segment was moved to a new issue that I can't find in [this](https://github.com/JanDeDobbeleer/oh-my-posh/issues/187#issuecomment-743762458) follow up.

### Environment

- Oh my Posh version: 3.96.0
- Theme: **modified** P10k modern
- Operating System: winodws, wsl1 (ubuntu 20.04) & Ubuntu 20.04
- Shell: powershell 7.1.3
- Terminal: windows terminal, vs code integrated terminal, & GNOME Terminal

### My Questions

 1. Does the `mapped_locations` property use the input string to match then map:
     1. Proper locations (absolute paths)?
     2. Directory/folder in the path that matches the string?
 2. Does the `mapped_locations` replace:
     1. The entire path with the resulting mapped string?
     2. The section of the path that matches the input string?
     3. Just the section of the path that would correspond with the directory/folder that the input string matches?

**Comment by lnu** [comment, marked as answer]
Right now, the `replaceMappedLocations` function checks if the current location **starts with** one of the prefix defined in mapped_locations and replaces it with a custom glyph/text.

Example: 
"mapped_locations": [
              ["D:\\dev", "\uf962"],
            ],

will replace d:\dev\ohmyposh to \uf962\ohmyposh

but you can't do something like this:
"mapped_locations": [
              ["dev", "\uf962"],
            ],

and expects that it will replace the text dev anywhere in the path.

I don't remember exactly why it wasn't done as we obviously discussed it in the issue related to ignore_folders. 

https://github.com/JanDeDobbeleer/oh-my-posh/blob/52b66d1e5be69b4fe3b7ff8270929d43b998d66e/src/segment_path.go#L187

**Comment by JanDeDobbeleer** [comment]
@lnu I tried but were too many side-effects we couldn't guarantee that to work as expected.
